src/hooks/useApi.js

```diff
@@ -0,0 +1,31 @@
+import { useState, useCallback, useEffect } from "react";
+
+export function useApi(asyncFunc, paramObj) {
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(null);
+  const [data, setData] = useState(null);
+
+  const wrrapedFunc = useCallback(
+    async (paramObj) => {
+      let result;
+      try {
+        setIsLoading(true);
+        setError(null);
+        result = await asyncFunc(paramObj);
+        setData(result);
```

Comment:
P1:
result 변수에 재할당할 일이 없으니 const로 선언하시는 걸 추천드려요.

```suggestion
      try {
        setIsLoading(true);
        setError(null);
        
        const result = await asyncFunc(paramObj);
        setData(result);
```